server/repository/src/db_diesel/name.rs

```diff
@@ -124,15 +124,76 @@ impl<'a> NameRepository<'a> {
 
         let result = final_query.load::<NameAndNameStoreJoin>(&self.connection.connection)?;
 
-        Ok(result.into_iter().map(to_domain).collect())
+        Ok(result.into_iter().map(Name::from_join).collect())
+    }
+
+    pub fn create_filtered_query(store_id: String, filter: Option<NameFilter>) -> BoxedNameQuery {
```

Comment:
```suggestion
    // create_filtered_query returns a list of names. If a `name_store_join` row does not exist for the specified `store_id`,  `name_store_join.id` will be null.
    pub fn create_filtered_query(store_id: String, filter: Option<NameFilter>) -> BoxedNameQuery {
```

Took me ages to figure out how this query is working. I think this comment might have helped a bit, so suggesting it.
Not sure why it's hard to understand, partly my understanding of what a name store join is, and partly I'm used to more focussed queries perhaps. I was expecting a `inner_join` on `store_id`  to only get visible names and exclude your own name which would make more sense in the context of this PR.

Reply:
> Took me ages to figure out how this query is working

Sorry, I would have commented on this more, but in the PR description i noted: `Name repository is used in this new repository, thus create filtered query is relocated to struct (no extra changes to that method)`, maybe I should have also made a comment in the PR that there is not changes to this method, this might have helped to ignore logic here and only see how it is used, which might also be a little bit confusing since the filter is applied in service (was applied in service now it's applied in program_supplier repository and reads like this:

`let name_filter = NameFilter::new().is_store(true).is_visible(false);`


> I was expecting a inner_join on store_id to only get visible names and exclude your own name which would make more sense in the context of this PR

Do you mean instead of using

I've updated comment downstream with: e06aef768f4964a77caf3621730f391514247e76, I still think it's not quite accurate, hopefully a little bit better

Reply:
New comment looks good thanks!